### PR TITLE
[Core-HttpClient-JDK][BUG] Remove Wrapping of UncheckedIOException

### PR DESCRIPTION
**Pull request by mssfang**
fixes: https://github.com/Azure/azure-sdk-for-java/issues/35692

### CODE REVIEW

**Review by lmolkova — APPROVED** [review]
Please consider adding a test so we don't break it in the future.

File: sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java
Comment (alzimmermsft) [inline sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java:78]: Why are we wrapping this when the API is checked with `IOException`? This seems to be wrapping needlessly.

Comment (alzimmermsft) [reply sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java:78]: I think this all should be changed to have the IOException be thrown without wrapping as an UncheckedIOException as that is breaking the contract defined in the Javadocs.

```java
/**
     * Transfers body bytes to the {@link WritableByteChannel}.
     * @param channel The destination {@link WritableByteChannel}.
     * @throws IOException When I/O operation fails.
     * @throws NullPointerException When {@code channel} is null.
     */
```

Comment (lmolkova) [reply sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java:78]: I didn't realize we never shipped stabIe jdkclient. In this case, I'm fully in favor of breaking change anywhere we already wrap  `IOException` (in methods that supposed to throw it)

Comment (alzimmermsft) [reply sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java:78]: If I'm going to guess what happened is when this was initially being implemented `LOGGER.logThrowableAsError(ioException)` caused Spotbugs to incorrectly flag this as a bug (https://github.com/spotbugs/spotbugs/issues/1219) so `LOGGER.logExceptionAsError` was used wrapping with an UncheckedIOException. Instead, the logging should happen on one line and then the exception gets thrown without wrapping (and suppress our Checkstyle rule for logging exceptions [though it shouldn't raise this as a bug but I think it does]).

**Comment by mssfang** [comment]
@alzimmermsft @lmolkova I added the test and CHANGELOG. Can I have another review before I merge it?

**Pull request merged**
